src/createServer.js

```diff
@@ -1,8 +1,43 @@
-'use strict';
+const http = require('http');
+const path = require('path');
+const fs = require('fs');
 
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = http.createServer(async (req, res) => {
+    const { pathname } = new URL(req.url || '', `http://${req.headers.host}`);
+    const requestedPath = pathname.replace('/file', '') || 'index.html';
```

Comment:
The logic here assumes that if `pathname.replace('/file', '')` results in an empty string, it defaults to 'index.html'. This might not be the intended behavior if the root path is accessed. Consider revisiting this logic to ensure it aligns with the expected routing behavior.